File: app/src/main/java/com/adsamcik/signalcollector/utility/Failure.java
package com.adsamcik.signalcollector.utility;

import android.support.annotation.NonNull;

public class Failure<T> {
	public final T value;

	/**
	 * Creates successful instance
	 */
	public Failure() {
		value = null;
	}

	/**
	 * Create unsuccessful instance
	 *
	 * @param message value
	 */
	public Failure(@NonNull T message) {
		this.value = message;
	}

	/**
	 * Returns failure
	 *
	 * @return true if failed
	 */
	public boolean hasFailed() {
		return value != null;
	}
}

File: base/src/main/java/com/adsamcik/signals/base/Failure.kt
package com.adsamcik.signals.base

class Failure<T> {
    val value: T?

    /**
     * Creates successful instance
     */
    constructor() {
        value = null
    }

    /**
     * Create unsuccessful instance
     *
     * @param message value
     */
    constructor(message: T) {
        this.value = message
    }

    /**
     * Returns failure
     *
     * @return true if failed
     */
    fun hasFailed(): Boolean = value != null
}
